### DISCUSSION TITLE
How to deal with padded bboxes

### DISCUSSION THREAD

**Discussion by austinmw** [kornia/kornia#1583] [Questions]
Hi, I'm using tensorcom to serve batched `(images,labels)` pairs. The labels are are padded bboxes with class ids, so their size is `(N, max_boxes, 5)` (where the 5 are x,y,w,h,c). 

For example, when `max_boxes=16`, `labels[0]` may look like:
```
tensor([[ 53.7225,  70.2449, 160.4200, 148.4616,   1.0000],
        [245.6480,  47.0510,  88.7575, 173.8273,   1.0000],
        [351.4485, 103.4458,  24.0240,  75.9474,   1.0000],
        [330.2715, 104.9999,  24.6740,  71.0577,   1.0000],
        [ 88.3610, 200.2463, 123.0775,  37.9672,  41.0000],
        [ 66.3325, 118.3102,  38.2395,  49.3918,   1.0000],
        [ 81.8675, 124.5265,  29.0745,  46.0041,   1.0000],
        [263.4060, 203.0228,  80.5480,  23.9677,  41.0000],
        [ 96.1025, 127.6086,  13.6955,  21.1391,   1.0000],
        [173.1405, 132.2382,  11.0500,  30.1904,   1.0000],
        [  0.0000,   0.0000,   0.0000,   0.0000,   0.0000],
        [  0.0000,   0.0000,   0.0000,   0.0000,   0.0000],
        [  0.0000,   0.0000,   0.0000,   0.0000,   0.0000],
        [  0.0000,   0.0000,   0.0000,   0.0000,   0.0000],
        [  0.0000,   0.0000,   0.0000,   0.0000,   0.0000],
        [  0.0000,   0.0000,   0.0000,   0.0000,   0.0000]], device='cuda:0')
```

What's the most efficient way to input these pairs into an image/bbox augmentation pipeline? Do the augmentations support padded bbox batches? I thought that if not, converting the tensor to a list of unpadded tensors might not be efficient.

**Comment by edgarriba** [comment]
Hi, have you checked out the `AugmentationSequential`  ? /cc @shijianjian 
https://kornia.readthedocs.io/en/latest/augmentation.container.html#augmentation-sequential

**Reply by austinmw** [reply]
Hi @edgarriba, I've checked that out, but I could not determine how to pass padded bboxes. When I attempt to do so I get an error that some boxes have zero width. It seems I need to split the batched tensor into a list and unpad each image's corresponding boxes, which slows down my dataloading pipeline.

**Reply by twsl** [reply]
One workaround could be to make the padded boxes valid boxes by setting the max values to the width and height of the image and later filter them out based on the class id.
